src/EditorFeatures/CSharp/StringCopyPaste/AbstractPasteProcessor.cs

```diff
@@ -26,13 +28,13 @@ internal abstract class AbstractPasteProcessor
         /// <summary>
         /// The buffer's snapshot prior to the paste application.
         /// </summary>
-        protected readonly ITextSnapshot SnapshotBeforePaste;
+        protected readonly ITextSnapshot2 SnapshotBeforePaste;
```

Comment:
ITextSnapshot2 allows us to easily fork a textsnapshot and see how it impacts tracking spans.  This was needed so we can map regions forward across potential speculative edits to see waht the text looks like after.